src/cast.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+import Raw from './raw';
+import {keys} from "lodash";
+
+class Cast extends Raw {
+  constructor(client) {
+    super(client);
+
+    this.value = null;
+    this.to    = '';
+    this.alias = '';
+  }
+
+  cast(value, to) {
+    this.value = value;
+    this.to    = to;
+
+    return this;
+  }
+
+  as(alias) {
+    this.alias = alias;
+
+    return this;
+  }
+
+  static get TYPES() {
+    return {
```

Comment:
Unfortunately it's not quite this generic across dialects. Will need a mapping per dialect.

Reply:
Is this the reason why PR was closed?

Reply:
Yes. Will need to rethink structure and research different dialects before-hand. I'm not keen on oracle/mssql syntax.

Will most likely end up having to create a base TYPES map on the client, and make deviations on each individual dialects' client.